packages/clerk-js/src/ui/components/OrganizationProfile/AddDomainForm.tsx

```diff
@@ -34,34 +45,51 @@ export const AddDomainPage = withCardStateProvider(() => {
     e.preventDefault();
     return organization
       .createDomain(nameField.value)
-      .then(res => {
+      .then(async res => {
+        setDomainId(res.id);
+        domains?.revalidate?.();
         if (res.verification && res.verification.status === 'verified') {
-          return navigate(`../domain/${res.id}?mode=select`);
+          onSuccess();
+        } else {
+          wizard.nextStep();
         }
-        return navigate(`../domain/${res.id}/verify`);
```

Comment:
@desiprisg are we good to be replacing these?

Reply:
yes